cmd/promtool/main.go

```diff
@@ -364,6 +382,48 @@ func (ls lintConfig) lintDuplicateRules() bool {
 	return ls.all || ls.duplicateRules
 }
 
+// Check server health.
+func CheckServerHealth(url string) error {
+	if url == "" {
+		url = "http://localhost:9090"
+	}
+
+	// Check Health
+	healthCheckResp, err := http.Get(url + "/-/healthy")
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "  FAILED:", err)
+		return err
+	}
+	healthCheckBody, err := io.ReadAll(healthCheckResp.Body)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "  FAILED:", err)
+		return err
+	}
+
+	fmt.Printf("  SUCCESS: %v", string(healthCheckBody))
+	return nil
+}
+
+// Check server readiness.
+func CheckServerReadiness(url string) error {
+	if url == "" {
+		url = "http://localhost:9090"
+	}
+	// Check Readiness
+	readinessCheckResp, err := http.Get(url + "/-/ready")
+	if err != nil {
```

Comment:
err is only connection errors, this probably needs to FAIL also if the response isn't 200.

(Although strangely while there's a testReady in web/web.go, it's not hooked up to /-/ready which somewhat surprises me... but that's maybe something to look at separately to the implementation of this command.)

Reply:
Addressed